src/operations/client_bulk_write/command_builder.ts

```diff
@@ -179,6 +222,25 @@ export class ClientBulkWriteCommandBuilder {
   }
 }
 
+function validateBufferSize(
+  name: string,
+  buffer: Uint8Array,
+  maxBsonObjectSize: number,
+  maxMessageSizeBytes: number
+) {
+  if (buffer.length > maxBsonObjectSize) {
+    throw new MongoInvalidArgumentError(
+      `Client bulk write operation ${name} of length ${buffer.length} exceeds the max bson object size of ${maxBsonObjectSize}`
+    );
+  }
+
+  if (buffer.length > maxMessageSizeBytes) {
```

Comment:
This is for the validation checks of individual ops not the total size for the prose tests. The total size checks happen on lines 145 and 182.

Reply:
What I'm wondering is if `maxMessageSizeBytes` is ever less than or equal to `maxBSONObjectSize`.  My gut says that `maxMessageSizeBytes` is strictly greater than `maxBSONObjectSize` because OP_MSG is, in the simplest case, a few bytes of overhead + a BSON document and the the document's length is always less than or equal to `maxBSONObjectSize`.

If that's the case, then we'll never enter this if statement because `buffer.length > maxMessageSizeBytes > maxBSONObjectSize`, so we'd throw from the first if-statement.

Reply:
Oh apologies I was quite clear on what you meant before, but I do now. I think your statement is correct in that I think it's impossible for `maxMessageSizeBytes` to ever be less than or equal to `maxBSONObjectSize` when using OP_MSG or any part of our wire protocol. This is just defensive programming in case in the future for some strange reason that could happen, and the spec prose tests explicitly called both out. I've removed the second check as I think you were suggesting to do.